### PR TITLE
ZEN-31153 Users with no permissions should not see any devices

### PR DESCRIPTION
**Pull request by romanbakaleyko**

### CODE REVIEW

**Comment by romanbakaleyko** [comment]
modified permissions in the way like we have in 6.x

File: Products/ZenModel/migrate/czadminrole.py [hunk 1]
```diff
@@ -68,8 +77,12 @@ def cutover(self, dmd):
...
                      for entry in zport.rolesOfPermission(perm)
                      if entry['selected']]
-            if ZEN_MANAGER_ROLE in roles:
-                zport.manage_permission(perm, roles + [CZ_ADMIN_ROLE], 1)
+            if ZEN_MANAGER_ROLE in roles:                
+                if perm not in NO_ACQUIRE_PERMS:
+                    zport.manage_permission(perm, roles + [CZ_ADMIN_ROLE], 1)
+                else:
+                    zport.manage_permission(perm, roles + [CZ_ADMIN_ROLE], 0)
+
```
Comment (jpeacock-zenoss) [inline Products/ZenModel/migrate/czadminrole.py]: Modifying existing migration scripts is not typically recommended, especially if it's already been released.  Any fixes should be applied in a later migration script.

File: Products/ZenModel/migrate/fixczadminrole.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+NO_ACQUIRE_PERMS = (ZEN_CHANGE_DEVICE_PRODSTATE,
+                    ZEN_CHANGE_EVENT_VIEWS,
+                    ZEN_MANAGE_EVENTS,
+                    ZEN_VIEW)
+
+
+class FixCZAdminRole(Migrate.Step):
+
+    version = Migrate.Version(300, 0, 5)
```
Comment (jpeacock-zenoss) [inline Products/ZenModel/migrate/fixczadminrole.py]: This is a new migration, so it should have a
```
from Products.ZenModel.ZMigrateVersion import SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION
```
and `version` should be:
```
version = Migrate.Version(SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION)
```

**Comment by romanbakaleyko** [comment]
WARNING! We have another PR with changes to zodb.sql.gz https://github.com/zenoss/zenoss-prodbin/pull/3519 don't merge them both!!!

We need to merge one and then for another pull changes for prodbin and re-run sql dump.

**Comment by romanbakaleyko** [comment]
retest

**Pull request merged**